src/main/java/com/list/todo/controllers/UserController.java

```diff
@@ -22,65 +28,39 @@ public UserController(UserService userService) {
-		if (users.isEmpty()){
-			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-		}
-		return new ResponseEntity<>(users, HttpStatus.OK);
-	}
-
-	@GetMapping(value = "/users/{id}", produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
-	public ResponseEntity<User> getUser(@PathVariable Long id) {
-		User user = userService.getUserById(id);
-		if (user == null){
-			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-		}
-		return new ResponseEntity<>(user, HttpStatus.OK);
-	}
-	
-	@PostMapping(value = "/users")
-	public ResponseEntity<Void> addUser(@RequestBody User user, UriComponentsBuilder ucBuilder) {
-		if (userService.isUserExist(user)){
-			return new ResponseEntity<>(HttpStatus.CONFLICT);
-		}
-
-		userService.saveUser(user);
+	@GetMapping("/me")
+	@PreAuthorize("hasRole('ROLE_USER')")
+    public UserSummary getCurrentUser(@CurrentUser UserPrincipal currentUser) {
+		User user = userService.getUserById(currentUser.getId());
 
-		HttpHeaders headers = new HttpHeaders();
-		headers.setLocation(ucBuilder.path("/users/{id}").buildAndExpand(user.getUserId()).toUri());
-		return new ResponseEntity<>(headers, HttpStatus.CREATED);
-	}
+		return new UserSummary(user.getId(), user.getUsername(), user.getName());
+    }
 	
-	@PutMapping(value = "/users/{id}")
-	public ResponseEntity<User> updateUser(@PathVariable Long id, @RequestBody User user) {
-		User currentUser = userService.getUserById(id);
+	@PutMapping("/editProfile")
+	@PreAuthorize("hasRole('ROLE_USER')")
+	public ResponseEntity<User> updateUser(@CurrentUser UserPrincipal currentUser, @RequestBody User user) {
+		User currUser = userService.getUserById(currentUser.getId());
 
-		if (currentUser == null){
+		if (currUser == null){
 			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
 		}
 
-		currentUser.setName(user.getName());
-		currentUser.setSurname(user.getSurname());
-		currentUser.seteMail(user.geteMail());
-		currentUser.setPassword(user.getPassword());
-		currentUser.setProjects(user.getProjects());
+		currUser.setName(user.getName());
+		currUser.setUsername(user.getUsername());
+		currUser.setEmail(user.getEmail());
+		currUser.setPassword(user.getPassword());
 
-		userService.updateUser(currentUser);
+		userService.updateUser(currUser);
 
-		return new ResponseEntity<>(currentUser, HttpStatus.OK);
+		return new ResponseEntity<>(currUser, HttpStatus.OK);
 	}
 	
-	@DeleteMapping(value = "/users/{id}")
-	public ResponseEntity<User> deleteUser(@PathVariable Long id) {
-		User user = userService.getUserById(id);
+	@DeleteMapping("/deleteProfile")
+	@PreAuthorize("hasRole('ROLE_USER')")
+	public ResponseEntity<User> deleteUser(@CurrentUser UserPrincipal currentUser) {
 
-		if (user == null){
-			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-		}
-		userService.getUserById(id);
+		userService.deleteUser(currentUser.getId());
 
-		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+		return new ResponseEntity<>(HttpStatus.OK);
```

Comment:
NO_CONTENT is standard response for DELETE

Reply:
We'll fix in next commit